### PR TITLE
LBAC-22 Added Unit tests for EncounterSearchAdviser

### PR DESCRIPTION
**Pull request by Bisht13**
## Description of what I changed
I've added unit tests for EncounterSearchAdviser named EncounterSearchAdviserTest.
In the test I've tested that all admin can access all users from different ways.
I've also given privileges to users as required.
## Issue I worked on
[Issue](https://issues.openmrs.org/browse/LBAC-22)
[Task Instance](https://codein.withgoogle.com/dashboard/task-instances/6518619409219584/)

## Checklist: I completed these to help reviewers :)
- [x] My pull request only contains **ONE single commit**

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

**Comment by Bisht13** [comment]
@suthagar23 can you check the first 3 tests for the method getEncounter(). I don't think they are working as intended.

File: api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java [hunk 1]
```diff
@@ -0,0 +1,1127 @@
...
+        Encounter encounter8 = encounterService.getEncounter(8);
+        assertNotNull(encounter8);
+        assertEquals(encounter8.getLocation(),location2);
+    }
+    
+    @Test
+    public void getEncounter_nonAdminUserWithOutLocationPropertyShouldNotGetEncounterInformation(){
+        User normalUser = userService.getUser(DEMO_USER1_ID);
+        Context.authenticate(normalUser.getUsername(), "userServiceTest");
+        assertFalse(normalUser.isSuperUser());
```
Comment (suthagar23) [inline api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java]: Here your test case indicated that it should be a normal user. Then why are you authenticating as an Admin (super user) here?

File: api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java [hunk 2]
```diff
@@ -0,0 +1,1127 @@
...
+    private static final int DEMO_LOCATION1_ID = 1;
+    private static final int DEMO_LOCATION2_ID = 2;
+    
+    public EncounterSearchAdviserTest() {
+    }
+    
+    @Override
+    protected void setInterceptorAndServices(TestWithAOP testCase) {
+        testCase.setInterceptor(UserServiceInterceptorAdvice.class);
+        testCase.addService(UserService.class);
```
Comment (suthagar23) [inline api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java]: You have connected with UserServiceInterceptorAdvice, UserService classes here, then how can you test for EncounterServiceInterceptorAdvice and EncounterService?

Comment (Bisht13) [reply api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java]: Made the necessary changes 😃

File: api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java [hunk 3]
```diff
@@ -0,0 +1,861 @@
...
+        assertNotNull(encounter6);
+        assertEquals(encounter6.getLocation(),location1);
+        assertNotNull(encounter7);
+        assertEquals(encounter7.getLocation(),location1);
+        assertNotNull(encounter8);
+        assertEquals(encounter8.getLocation(),location2);
+    }
+    
+    @Test
+    public void getEncounters_nonAdminUserWithOutLocationPropertyShouldNotGetEncounterInformation(){
```
Comment (suthagar23) [inline api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java]: Have you checked for null user property?

File: api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java [hunk 4]
```diff
@@ -0,0 +1,861 @@
...
+        assertNotNull(encounter7);
+        assertEquals(encounter7.getLocation(),location2);
+        
+        Encounter encounter8 = encounterService.getEncounterByUuid("0619d653-393b-4118-9c83-a3715b82d4ac");
+        assertNotNull(encounter8);
+        assertEquals(encounter8.getLocation(),location2);
+    }
+    
+    @Test
+    public void getEncounterByUuid_nonAdminUserWithOutLocationPropertyShouldNotGetEncounterInformation(){
```
Comment (suthagar23) [inline api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java]: Have you checked for null user property?

File: api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java [hunk 5]
```diff
@@ -0,0 +1,861 @@
...
+        assertNotNull(encounter6);
+        assertEquals(encounter6.getLocation(),location2);
+        assertNotNull(encounter7);
+        assertEquals(encounter7.getLocation(),location1);
+        assertNotNull(encounter8);
+        assertEquals(encounter8.getLocation(),location1);
+    }
+            
+    @Test
+    public void getEncountersByPatientId_nonAdminUserWithOutLocationPropertyShouldNotGetEncounterInformation(){
```
Comment (suthagar23) [inline api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java]: Have you checked for null user property?

File: api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java [hunk 6]
```diff
@@ -0,0 +1,861 @@
...
+        assertNotNull(encounter6);
+        assertEquals(encounter6.getLocation(),location1);
+        assertNotNull(encounter7);
+        assertEquals(encounter7.getLocation(),location1);
+        assertNotNull(encounter8);
+        assertEquals(encounter8.getLocation(),location2);
+    }
+    
+    @Test
+    public void getEncountersByPatient_nonAdminUserWithOutLocationPropertyShouldNotGetEncounterInformation(){
```
Comment (suthagar23) [inline api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java]: Have you checked for null user property?

File: api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java [hunk 7]
```diff
@@ -0,0 +1,861 @@
...
+        Visit visit2 = visitService.getVisit(2);
+        List<Encounter> encounterList2 = encounterService.getEncountersByVisit(visit2,false);
+        assertEquals(encounterList2.size(),1);
+        Encounter encounter3 = encounterList2.get(0);
+        assertNotNull(encounter3);
+        assertEquals(encounter3.getLocation(),location2);
+    }
+    
+    @Test
+    public void getEncountersByVisit_nonAdminUserWithOutLocationPropertyShouldNotGetEncounterInformation(){
```
Comment (suthagar23) [inline api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java]: Have you checked for null user property?

File: api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java (hunk 5 above)
Comment (suthagar23) [reply api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java]: Ask this question for rest of the test cases as well

File: api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java (hunk 2 above)
Comment (suthagar23) [reply api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java]: Can you confirm that all functionalities of the module class are working as expected?

Comment (Bisht13) [reply api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java]: Yes @suthagar23 the module class is working as intended 😄.

File: api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (Bisht13) [reply api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java]: Fixed 😃

File: api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java (hunk 3 above)
Comment (Bisht13) [reply api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java]: Fixed 😃

File: api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java (hunk 4 above)
Comment (Bisht13) [reply api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java]: Fixed 😃

File: api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java (hunk 5 above)
Comment (Bisht13) [reply api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java]: Fixed 😃

File: api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java (hunk 6 above)
Comment (Bisht13) [reply api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java]: Fixed 😃

File: api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java (hunk 7 above)
Comment (Bisht13) [reply api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java]: Fixed 😃

**Comment by suthagar23** [comment]
Can you check for the Travis failure?

**Comment by Bisht13** [comment]
> Can you check for the Travis failure?

Fixed it. The Travis Build is now passing.

**Comment by suthagar23** [comment]
Thanks for your great work @Bisht13

File: api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java [hunk 8]
```diff
@@ -0,0 +1,441 @@
...
+        User authenticatedUser = Context.getAuthenticatedUser();
+        assertNotNull(authenticatedUser);
+        assertTrue(authenticatedUser.isSuperUser());
+        
+        Location location1 = locationService.getLocation(DEMO_LOCATION1_ID);
+        Location location2 = locationService.getLocation(DEMO_LOCATION2_ID);
+        
+        Encounter encounter3 = encounterService.getEncounter(3);
+        assertNotNull(encounter3);
+        assertEquals(encounter3.getLocation(),location1);
```
[the remark this answers is not in the document]
Comment (Bisht13) [reply api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java]: Changed all tests accordingly.

**Pull request merged**